libs/designer/src/lib/ui/mcp/connectors/ListConnectors.tsx

```diff
@@ -0,0 +1,233 @@
+        displayName: metadata?.connectorTitle,
+        iconUri: metadata?.iconUri,
+        connectionName,
+        isConnected,
+      });
+
+      return acc;
+    }, []);
+  }, [connectionReferences, connectionsMapping, operationInfos, operationMetadata]);
+
+  const columns = [
+    { columnKey: 'connector', label: INTL_TEXT.connectorLabel },
+    { columnKey: 'connection', label: INTL_TEXT.connectionLabel },
+    { columnKey: 'status', label: INTL_TEXT.statusLabel },
+    { columnKey: 'actions', label: '' }, // Empty label for actions column
+  ];
+
+  const handleEditConnector = useCallback(
+    (connectorId: string) => {
+      dispatch(
+        openConnectorPanelView({
+          panelView: McpPanelView.SelectConnector,
+        })
+      );
+      dispatch(selectConnectorId(connectorId));
+      dispatch(selectOperations([]));
+    },
+    [dispatch]
+  );
+
+  const handleDeleteConnector = useCallback(
+    (connectorId: string) => {
+      const operationIdsToDelete = Object.entries(operationInfos)
+        .filter(([_, info]) => info?.connectorId === connectorId)
+        .map(([operationId, _]) => operationId);
+
+      if (operationIdsToDelete.length > 0) {
+        dispatch(deinitializeNodes(operationIdsToDelete));
+        dispatch(deinitializeOperationInfos({ ids: operationIdsToDelete }));
+      }
+    },
+    [operationInfos, dispatch]
+  );
+
+  if (!items || items.length === 0) {
+    return (
+      <div className={styles.emptyState}>
+        <div className={styles.emptyStateIcon}>
+          <ConnectorFilled />
+        </div>
+        <Text size={400} weight="semibold" style={{ marginBottom: '8px' }}>
+          {INTL_TEXT.noConnectors}
+        </Text>
+        <Text size={200} style={{ opacity: 0.7, marginBottom: '24px' }}>
+          {INTL_TEXT.addFirstConnector}
+        </Text>
+      </div>
+    );
+  }
+
+  return (
+    <Table className={styles.tableStyle} aria-label={INTL_TEXT.tableAriaLabel} size="small">
+      <TableHeader>
+        <TableRow style={connectorTableCellStyles}>
+          {columns.map((column) => (
+            <TableHeaderCell key={column.columnKey} style={connectorTableCellStyles}>
+              <Text weight="semibold">{column.label}</Text>
+            </TableHeaderCell>
+          ))}
+        </TableRow>
+      </TableHeader>
+      <TableBody style={connectorTableCellStyles}>
+        {items.map((item) => (
+          <TableRow key={item.connectorId} style={connectorTableCellStyles}>
+            <TableCell className={styles.iconTextCell} style={connectorTableCellStyles}>
+              <img className={styles.connectorIcon} src={item.iconUri ?? DefaultIcon} alt={`${item.displayName} icon`} />
+              <Link as="button" onClick={() => handleEditConnector(item.connectorId)}>
+                {item.displayName}
+              </Link>
+            </TableCell>
+            <TableCell style={connectorTableCellStyles}>{item?.connectionName}</TableCell>
```

Comment:
Again this value needs to come from query cache... it is the same connection for everyone so either you can get it from cache in this component and use it, or do this as a component <ConnectionName connectionId={connectionId} /> and this component can do useConnection and get the display name from there